Dockerfile

```diff
@@ -35,41 +32,81 @@ LABEL summary="$SUMMARY" \
       version="1" \
       maintainer="Red Hat Insights"
 
-USER root
 
-RUN yum module reset nodejs -y
-RUN yum install @nodejs:14 -y
+# Very minimal set of packages
+# glibc-langpack-en is needed to set locale to en_US and disable warning about it
+# gcc to compile some python packages (e.g. ciso8601)
+# shadow-utils to make useradd available
+RUN INSTALL_PKGS="python38 python38-devel glibc-langpack-en libpq-devel gcc shadow-utils" && \
+    microdnf --nodocs -y upgrade && \
+    microdnf -y --setopt=tsflags=nodocs --setopt=install_weak_deps=0 install $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    microdnf -y clean all --enablerepo='*'
+
+# PIPENV_DEV is set to true in the docker-compose allowing
+# local builds to install the dev dependencies
+ARG PIPENV_DEV=False
+ARG USER_ID=1001
+
+# Create a Python virtual environment for use by any application to avoid
+# potential conflicts with Python packages preinstalled in the main Python
+# installation.
+RUN python3.8 -m venv /pipenv-venv
+ENV PATH="/pipenv-venv/bin:$PATH"
+# Install pipenv into the virtual env
+RUN \
+    pip install --upgrade pip && \
+    pip install pipenv
+
+WORKDIR ${APP_ROOT}
+
+# install dependencies
+ENV PIP_DEFAULT_TIMEOUT=100
+COPY Pipfile .
+COPY Pipfile.lock .
+RUN \
+    # install the dependencies into the working dir (i.e. ${APP_ROOT}/.venv)
+    pipenv install --deploy && \
+    # delete the pipenv cache
+    pipenv --clear
+
+
+# Runtime env variables:
+ENV VIRTUAL_ENV=${APP_ROOT}/.venv
+ENV \
+    # Add the rbac virtual env bin to the front of PATH.
+    # This activates the virtual env for all subsequent python calls.
+    PATH="$VIRTUAL_ENV/bin:$PATH" \
+    PROMETHEUS_MULTIPROC_DIR=/tmp
+
+# copy the src files into the workdir
+COPY . .
 
-# Copy the S2I scripts from the specific language image to $STI_SCRIPTS_PATH.
-COPY openshift/s2i/bin $STI_SCRIPTS_PATH
+RUN touch ${APP_HOME}/app.log; chmod 777 ${APP_HOME}/app.log
 
-# Copy extra files to the image.
-COPY openshift/root /
+# test
+RUN chown -R 1001:0 ${APP_ROOT}
 
-# Copy application files to the image.
-COPY . ${APP_ROOT}/src
+# create the rbac user
+RUN \
+    adduser rbac -u ${USER_ID} -g 0 && \
+    chmod ug+rw ${APP_ROOT} ${APP_HOME} ${APP_HOME}/static /tmp
+USER rbac
 
-# - Create a Python virtual environment for use by any application to avoid
-#   potential conflicts with Python packages preinstalled in the main Python
-#   installation.
-# - In order to drop the root user, we have to make some directories world
-#   writable as OpenShift default security model is to run the container
-#   under random UID.
-RUN pip install virtualenv && \
-    virtualenv ${APP_ROOT} && \
-    chown -R 1001:0 ${APP_ROOT} && \
-    fix-permissions ${APP_ROOT} -P && \
-    rpm-file-permissions && \
-    $STI_SCRIPTS_PATH/assemble || true
 
-RUN curl -L -o /usr/bin/haberdasher \
```

Comment:
great to see this - there is also lot of environment variables with name `HABERDASHER` which needs to be removed also - there are not needed now(+ app interface in `data/services/insights/rbac/deploy-clowder.yml`) - but you can do it in follow up PR.